comp/core/statsd/statsd.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package statsd
+
+import (
+	"net"
+	"os"
+	"strconv"
+
+	"go.uber.org/fx"
+
+	ddgostatsd "github.com/DataDog/datadog-go/v5/statsd"
+
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// Module defines the fx options for this component.
+var Module = fxutil.Component(
+	fx.Provide(newStatsdService),
+)
+
+type service struct{}
+
+// Get returns a pre-configured statsd client
+func (hs *service) Get(options ...ddgostatsd.Option) (ddgostatsd.ClientInterface, error) {
+	return GetClient("", options...)
+}
+
+// GetForAddr returns a pre-configured statsd client that defaults to `addr` if no env var is set
+func (hs *service) GetForAddr(addr string, options ...ddgostatsd.Option) (ddgostatsd.ClientInterface, error) {
+	return GetClient(addr, options...)
+}
+
+// GetForHostPort returns a pre-configured statsd client that defaults to `host:port` if no env var is set
+func (hs *service) GetForHostPort(host string, port int, options ...ddgostatsd.Option) (ddgostatsd.ClientInterface, error) {
+	return GetClient(net.JoinHostPort(host, strconv.Itoa(port)), options...)
+}
+
+var _ Component = (*service)(nil)
+
+// GetClient returns a pre-configured statsd client that defaults to `addr` if no env var is set
+// It is exported for callers that might not support components.
+func GetClient(addr string, options ...ddgostatsd.Option) (ddgostatsd.ClientInterface, error) {
```

Comment:
What is the reason for `GetClient` to be exported (It starts with an upper case)?

Reply:
I though it would be useful to have for parts that don't have access to components, but I removed it now